test/new_relic/agent/error_filter_test.rb

```diff
@@ -0,0 +1,96 @@
+# encoding: utf-8
```

Comment:
Oh also, we should add a few tests for the ignore and expect methods to confirm those work as expected, since from what i can tell, I don't think those are covered by these.